File: csharp/LogicAndTrick.WikiCodeParser/Elements/MdLineElement.cs
using LogicAndTrick.WikiCodeParser.Nodes;

namespace LogicAndTrick.WikiCodeParser.Elements
{
    public class MdLineElement : Element
    {
        public override bool Matches(Lines lines)
        {
            var value = lines.Value().TrimEnd();
            return value.Length >= 3 && value == new string('-', value.Length);
        }

        public override INode Consume(Parser parser, ParseData data, Lines lines, string scope)
        {
            return new HtmlNode("<hr />", PlainTextNode.Empty, "");
        }
    }
}

File: php/src/Elements/MdLineElement.php
<?php

namespace LogicAndTrick\WikiCodeParser\Elements;

use LogicAndTrick\WikiCodeParser\Lines;
use LogicAndTrick\WikiCodeParser\Nodes\HtmlNode;
use LogicAndTrick\WikiCodeParser\Nodes\INode;
use LogicAndTrick\WikiCodeParser\Nodes\PlainTextNode;
use LogicAndTrick\WikiCodeParser\ParseData;
use LogicAndTrick\WikiCodeParser\Parser;

class MdLineElement extends Element
{
    public function Matches(Lines $lines): bool
    {
        $value = rtrim($lines->Value());
        return strlen($value) >= 3 && $value == str_repeat('-', strlen($value));
    }

    public function Consume(Parser $parser, ParseData $data, Lines $lines, string $scope): ?INode
    {
        $ret = new HtmlNode('<hr />', PlainTextNode::Empty(), '');
        $ret->plainBefore = '---';
        return $ret;
    }
}